core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -1357,27 +1344,27 @@ private void testPositionDeletesBaseTableFilter(boolean transactional) {
         (StructLike)
             constantsMap(posDeleteTask, partitionType).get(MetadataColumns.PARTITION_COLUMN_ID);
     int taskPartition = taskPartitionStruct.get(0, Integer.class);
-    Assert.assertEquals("Expected correct partition on task's file", 0, filePartition);
-    Assert.assertEquals("Expected correct partition on task's column", 0, taskPartition);
-
-    Assert.assertEquals(
-        "Expected correct partition spec id on task", 0, posDeleteTask.file().specId());
-    Assert.assertEquals(
-        "Expected correct partition spec id on constant column",
-        0,
-        constantsMap(posDeleteTask, partitionType).get(MetadataColumns.SPEC_ID.fieldId()));
-
-    Assert.assertEquals(
-        "Expected correct delete file on task", FILE_A_DELETES.path(), posDeleteTask.file().path());
-    Assert.assertEquals(
-        "Expected correct delete file on constant column",
-        FILE_A_DELETES.path(),
-        constantsMap(posDeleteTask, partitionType).get(MetadataColumns.FILE_PATH.fieldId()));
+    assertThat(filePartition).as("Expected correct partition on task's file").isEqualTo(0);
+    assertThat(taskPartition).as("Expected correct partition on task's column").isEqualTo(0);
+
+    assertThat(posDeleteTask.file().specId())
+        .as("Expected correct partition spec id on task")
+        .isEqualTo(0);
+    assertThat(constantsMap(posDeleteTask, partitionType).get(MetadataColumns.SPEC_ID.fieldId()))
```

Comment:
this should be using `.containsEntry()`. Same for the other places in this PR

Reply:
Update the map tests with `.containsEntry()`. However each value is specified with the wildcard type like `capture of ?`, so the `contantsMap` value type needs to be changed in each test.

Reply:
If there's a better idea, please let me know.